### PR TITLE
add support for HH:mm:ss

### PR DESCRIPTION
**Pull request by rupal-bq**
*Issue:* #849

*Description of changes:*
 - added STRICT_DATE_FORMATTER

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by chloe-zh — APPROVED** [review]
LGTM, thanks for the fix!

**Review by penghuo — COMMENTED** [review]
Please add UT to cover the STRICT_DATE_FORMATTER

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java [hunk 1]
```diff
@@ -77,6 +78,7 @@
       new DateTimeFormatterBuilder()
           .appendOptional(SQL_LITERAL_DATE_TIME_FORMAT)
           .appendOptional(STRICT_DATE_OPTIONAL_TIME_FORMATTER)
+              .appendOptional(STRICT_DATE_FORMATTER)
```
Comment (dai-chen) [inline elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java:81]: Should `STRICT_HOUR_MINUTE_SECOND_FORMATTER` be registered here too?

Comment (rupal-bq) [reply elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprValueFactory.java]: removed STRICT_DATE_FORMATTER as the issue was fixed with STRICT_HOUR_MINUTE_SECOND_FORMATTER.

**Comment by rupal-bq** [comment]
@penghuo @dai-chen this change fixed the issue with calcs data but I was wondering if I should create `constructTime` similar to `constructTimestamp` instead ?

**Review by penghuo — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]
Thanks for the fix!

**Pull request merged**